### PR TITLE
Add table `aws_appautoscaling_policy`

### PR DESCRIPTION
**Pull request by jramosf**
This Pull Request adds the table `aws_appautoscaling_policy`, in a very similar implementation as the existing `aws_appautoscaling_target`. See: https://docs.aws.amazon.com/cli/latest/reference/application-autoscaling/describe-scaling-policies.html and https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/applicationautoscaling for more details.

- Additionally, I fixed an invalid reference to `aws_neptune_db_cluster` that I found in the existing `aws_appautoscaling_table`.

# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A (tests added in PR with Terraform and executed in pipeline)
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select policy_name, resource_id, policy_type from aws_appautoscaling_policy where service_namespace = 'ecs';
+------------------------+-------------------------------+-----------------------+
| policy_name            | resource_id                   | policy_type           |
+------------------------+-------------------------------+-----------------------+
| my-example-policy2     | service/my-cluster/my-service | StepScaling           |
| my-example-policy7     | service/my-cluster/my-service | TargetTrackingScaling |
+------------------------+-------------------------------+-----------------------+
>
```
</details>

### CODE REVIEW

**Comment by jramosf** [comment]
Thanks for the review @misraved . I've removed the `Get` function as AWS only implements `DescribeScalingPolicies` and both functions were calling the same AWS API.

**Comment by jramosf** [comment]
@ParthaI is there a change this gets merged? Thanks!

File: aws/table_aws_appautoscaling_policy.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+//// LIST FUNCTION
+
+func listAwsApplicationAutoScalingPolicies(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	name := d.EqualsQuals["service_namespace"].GetStringValue()
+
+	// Create Session
+	svc, err := ApplicationAutoScalingClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_appautoscaling_policy.listAwsApplicationAutoScalingPolicies", "get_client_error", err)
```
Comment (ParthaI) [inline aws/table_aws_appautoscaling_policy.go:113]:
```suggestion
		plugin.Logger(ctx).Error("aws_appautoscaling_policy.listAwsApplicationAutoScalingPolicies", "client_error", err)
```

**Comment by jramosf** [comment]
Thanks @ParthaI , committed suggested change and ready-to-review

File: docs/tables/aws_appautoscaling_policy.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  policy_type,
+  resource_id,
+  creation_time
+from
+  aws_appautoscaling_policy
+where
+  service_namespace = 'ecs';
+```
+
+
```
Comment (misraved) [inline docs/tables/aws_appautoscaling_policy.md:22]:
```suggestion
```

**Review by misraved — COMMENTED** [review]
Thanks @jramosf once again for this new table 👍. The structure looks great and it is very well implemented.

Could you please add 2-3 more example queries for the table? You could use the following headers for the example queries:
- List policies created in the last 30 days
- Get the CloudWatch alarms associated with the autoscaling policy (For this, it would be great if you could expand the `alarms` column of the table)
-  Get the step-tracking configuration of the autoscaling policies

**Review by misraved — APPROVED** [review]

**Comment by misraved** [comment]
Thanks @jramosf for making the changes so quickly. Great to have this table 👍.

**Pull request merged**